salt/utils/subprocess.py

```diff
@@ -4,12 +4,50 @@
 '''
 from __future__ import absolute_import, print_function, unicode_literals
 
+import os
+import sys
 import shlex
+import logging
 import subprocess
 import threading
 import salt.exceptions
-import salt.utils.data
 from salt.ext import six
+from salt.ext.six.moves import range
+
+
+log = logging.getLogger(__name__)
+
+
+class FdPopen(subprocess.Popen):
+    '''
+    Python2 closing file descriptors in a blind way by a system's given range.
+    Therefore close_fds can significantly slow down the system in various circumstances.
+    '''
+
+    if sys.platform.startswith('linux') and six.PY2:
+        def _close_fds(self, but):
+            '''
+            Close only file descriptors that are needed.
+            The issue is not new: https://bugs.python.org/issue1663329
+
+            Do *not* use os.closerange (!)
+
+            :param but:
+            :return:
+            '''
+            proc_path = '/proc/{}/fd'.format(os.getpid())
```

Comment:
See also https://stackoverflow.com/questions/899038/getting-the-highest-allocated-file-descriptor?answertab=active#tab-top